Add instructions for installing with conda

Boost is probably most easily installed with conda. Should include instructions for how to build against these libraries. 

Recently raised by @adamrkob,

> Hi Will,
>
> Sorry to bother again! But I am now working on new machines, and attempting to get it working has put me in the same position as before. A few things have changed for this round, though. Notable is that I am using conda for all installations, and am running python 3.6.
>
> I assume the problem is that I now need to point scans to the relevant directories for where conda puts things.
>
>```
> $ scons --includepath=/usr/local/include,/usr/include/malloc --libpath=/usr/local/lib
>```
>(or similar) worked before, but I am trying to determine what the correct paths are for the conda installed version. If you know these off the top of your head, feel free to let me know. Otherwise, I am using this in hopes of providing a record of the solution for the next time it comes up.
>
>Thanks,
>
>Adam

Sounds good Will,

OK, so here is where it is currently at. I went through with installing all the relevant packages (python, numpy, matplotlib, seaborn, scons and boost via conda), and using the Xcode/came-with-macos gcc, and verifying that the command line tools were installed. When attempting to run scons (note: MacOS with python 3.6.6 via anaconda3), I got the error listed in #20 that malloc.h could not be found:

> g++ -o Radiation_Model/source/OpticallyThick/OpticallyThickIon.o -c -std=c++11 -O3 -I/opt/local/include Radiation_Model/source/OpticallyThick/OpticallyThickIon.cpp
Radiation_Model/source/OpticallyThick/OpticallyThickIon.cpp:15:20: fatal error: malloc.h: No such file or directory
> #include <malloc.h>'

This was changed slightly, but not fully corrected by using:

`scons --includepath=$HOME/anaconda3/include --libpath=$HOME/anaconda3/lib`

SO, I tested commenting out the calls to malloc.h from :
'Radiation_Model/source/OpticallyThick/OpticallyThickIon.cpp
Radiation_Model/source/element.cpp
rsp_toolkit/source/fitpoly.cpp'

(using the call listed above). This allowed me to get further through the scons process, but it then broke on:

`g++ -o bin/ebtel++.run Radiation_Model/source/radiation.o Radiation_Model/source/ionfrac.o Radiation_Model/source/element.o Radiation_Model/source/OpticallyThick/OpticallyThickIon.o rsp_toolkit/source/fitpoly.o rsp_toolkit/source/xmlreader.o rsp_toolkit/source/file.o rsp_toolkit/tinyxml2/tinyxml2.o source/loop.o source/dem.o source/observer.o source/heater.o source/main.o -L/Users/akobelsk/anaconda3/lib -lboost_program_options-mt`
`ld: library not found for -lboost_program_options-mt
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

Changing line 35 of SConstruct by dropping the `-mt` flag so that it reads (in the base ebtelPlusPlus directory): 
`    env.Append(LIBS=['boost_program_options'])`
remedied this issue, and let scons seemingly run to completion. Unfortunately, it didn't give a proper output that ebtel++ could run:

`$ bin/ebtel++.run `
`dyld: Library not loaded: @rpath/libboost_program_options.dylib`
  `Referenced from: /Users/akobelsk/ebtelPlusPlus/bin/ebtel++.run`
  `Reason: image not found`
`Abort trap: 6`

I have just installed the conda gcc compiler, lets see if it fixes things.

--- a/README.md
+++ b/README.md
@@ -15,14 +15,8 @@ If you use ebtel++ in any published work, please cite the following papers:
 
 ## Installation
 
-Download the dependencies,
-
-* [git](https://git-scm.com/)
-* [gcc](https://gcc.gnu.org/) (v4.7 or later)
-* [scons](http://scons.org/)
-* [boost](http://www.boost.org/) (v1.53 or later)
-
-Then to install, compile, and run the code using the example configuration file in `config/ebtel.example.cfg.xml`,
+First, [make sure you have the needed dependencies](http://rice-solar-physics.github.io/ebtelPlusPlus/#dependencies).
+Then, clone, compile, and run the code using the example configuration file in `config/ebtel.example.cfg.xml`,
 
 ```Shell
 $ git clone https://github.com/rice-solar-physics/ebtelPlusPlus.git

--- a/docs/mkdocs/index.md
+++ b/docs/mkdocs/index.md
@@ -10,9 +10,18 @@ EBTEL also calculates the differential emission measure (DEM) for both the trans
 
 To compile ebtel++, first install the following dependencies,
 
-* [gcc](https://gcc.gnu.org/) (at least v4.7; included with OS X, most Linux distros; [cygwin](https://www.cygwin.com/) on Windows)
-* [scons](http://scons.org/) (`pip install scons` via [PyPI](https://pypi.python.org/pypi) or `conda install scons` with [Anaconda](https://www.continuum.io/downloads))
-* [boost](http://www.boost.org/) (at least v1.53; `sudo apt-get install libboost-dev` on Debian Linux, `sudo port install boost` via [Macports](https://www.macports.org/) on OS X, [from source](https://github.com/rice-solar-physics/IonPopSolver#installing-boost-from-source) on Windows; can also be installed via [Anaconda](https://www.continuum.io/downloads))
+* [gcc](https://gcc.gnu.org/) (at least v4.7)
+    * included with OS X as well as most Linux distributions
+    * [cygwin](https://www.cygwin.com/) on Windows
+* [scons](http://scons.org/)
+    * `pip install scons` with [PyPI](https://pypi.python.org/pypi)
+    * `conda install scons` with [Anaconda](https://www.continuum.io/downloads)
+* [boost](http://www.boost.org/) (at least v1.53)
+    * `conda install boost` with [Anaconda](https://www.continuum.io/downloads)
+    * `brew install boost` with [Homebrew](https://formulae.brew.sh/formula/boost)
+    * `sudo port install boost` with [Macports](https://ports.macports.org/port/boost/)
+    * `sudo apt-get install libboost-all-dev` on Debian Linux
+    * [from source](https://www.boost.org/doc/)
 
 Additionally, if you'd like to run the included tests and examples, you'll need the following Python dependencies, all easily installed with [anaconda](https://www.continuum.io/downloads),
 
@@ -31,14 +40,6 @@ $ cd ebtelPlusPlus
 $ scons
 ```
 
-If the compile step fails because the compiler cannot find the appropriate headers and/or libraries using the default locations, you can use the `--includepath` and/or `--libpath` flags, respectively. For example, if you've installed the Boost headers and libraries in `/usr/local/include` and `/usr/local/lib`,
-
-```Shell
-$ scons --includepath=/usr/local/include --libpath=/usr/local/lib
-```
-
-For more information about the available flags that can be passed to `scons`, you can run `scons -h`.
-
 This will create an executable `bin/ebtel++.run`. To see the available command line parameters,
 
 ```Shell
@@ -51,9 +52,45 @@ and to run the executable with the default configuration file `config/ebtel.exam
 $ bin/ebtel++.run
 ```
 
+### A Note on Compiling against the `boost` Libraries
+
+The included `scons` configuration will do it's best to guess where to find the needed `boost` headers and how to link against the needed libraries.
+However, depending on your operating system and or installation method, the locations and names of these libraries can vary.
+As such, you may find that, when compiling using `scons`, you get errors like,
+
+```Shell
+ld: library not found for -lboost_program_options-mt
+```
+
+Similarly, you may find that you get runtime errors about not being able to find the needed libraries when running the executable such as,
+
+```Shell
+dyld[21052]: Library not loaded: '@rpath/libboost_program_options.dylib'
+  Referenced from: '/home/user/ebtelPlusPlus/bin/ebtel++.run'
+  Reason: tried: '/usr/local/lib/libboost_program_options.dylib' (no such file), '/usr/lib/libboost_program_options.dylib' (no such file)
+[1]    21052 abort      bin/ebtel++.run
+```
+
+To resolve these issues, there are a number of optional flags that you can pass to `scons` when compiling the executable.
+For more information about the available flags, you can run `scons -h`.
+Additionally, [this GitHub issue](https://github.com/rice-solar-physics/ebtelPlusPlus/issues/74) may be helpful when debugging compile-time or runtime issues related to correctly compiling and linking against `boost`.
+
+As an example, if you have installed `boost` with Anaconda in the environment `ebtel-env` and your Anaconda installation is located at `/home/user/anaconda`, then you will likely need to pass the following flags to `scons`,
+
+```Shell
+$ scons \
+   --includepath=/home/user/anaconda/envs/ebtel-env/include \
+   --libpath=/home/user/anaconda/envs/ebtel-env/lib \
+   --linkflags="-rpath /home/user/anaconda/envs/ebtel-env/lib" \
+   --libs=boost_program_options
+```
+
+If instead you've installed `boost` in the base Anaconda environment, then you can just drop the `envs/ebtel-env` portion of each of the above paths.
+Note that the line breaks are only for readability and are not actually necessary when you compile your code.
+
 ## Testing
 
-If you've installed the above Python dependencies, you can also run the tests using,
+To install the needed Python dependencies and run the tests,
 
 ```Shell
 $ pip install -r requirements/requirements-test.txt

